src/Login.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react'
+import {
+  Grid,
+  Row,
+  Col,
+  FormGroup,
+  ControlLabel,
+  FormControl,
+  HelpBlock,
+  Button,
+  Alert
+} from 'react-bootstrap'
+import { withRouter } from 'react-router'
+
+import NavBar from './components/NavBar'
+import { Api } from './utils'
+
+function FieldGroup({ id, label, help, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+      {help && <HelpBlock>{help}</HelpBlock>}
+    </FormGroup>
+  )
+}
+
+class Login extends Component {
+  state = {
+    error: null
+  }
+
+  onSubmitForm = e => {
+    e.preventDefault()
+    const { username, password } = e.target
+    const data = {
+      username: username.value,
+      password: password.value
+    }
+
+    // TODO: replace URL with correct url
+    Api.postLogin(data)
+      .then(({ message: error, status }) => {
+        this.props.history.push('/', { isEditing: true })
+        if (status === 'failed') {
+          this.setState({ error })
+        } else {
+          this.props.history.push('/', { isEditing: true })
```

Comment:
It's probably better to set some state like `isLoggedIn` and then have the option to go into edit mode. Although the user will usually login just to be able to go into edit mode, it's not the best user experience to assume that and have login directly lead to edit mode